### PR TITLE
Expose scrolling values & add scrollToPixel capabilities

### PR DESCRIPTION
**Pull request by JordanMartinez**

### CODE REVIEW

File: src/main/java/org/fxmisc/flowless/VirtualFlowContent.java [hunk 1]
```diff
@@ -177,6 +177,44 @@ void scrollY(double deltaY) {
...
+    Val<Double> estimatedWidthPositionProperty() {
+        switch (getContentBias()) {
+            case HORIZONTAL: // vertical flow
+                return breadthPositionEstimateProperty();
+            case VERTICAL: // horizontal flow
+                return lengthPositionEstimateProperty();
+            default:
+                throw new AssertionError("Unreachable code");
+        }
+    }
```
Comment (TomasMikula) [inline src/main/java/org/fxmisc/flowless/VirtualFlowContent.java]: Would you mind rewriting this method as

    orientation.horizontalPositionProperty(content)

File: src/main/java/org/fxmisc/flowless/VirtualFlowContent.java [hunk 2]
```diff
@@ -177,6 +177,44 @@ void scrollY(double deltaY) {
...
+    Val<Double> estimatedHeightPositionProperty() {
+        switch (getContentBias()) {
+            case HORIZONTAL: // vertical flow
+                return lengthPositionEstimateProperty();
+            case VERTICAL: // horizontal flow
+                return breadthPositionEstimateProperty();
+            default:
+                throw new AssertionError("Unreachable code");
+        }
+    }
```
Comment (TomasMikula) [inline src/main/java/org/fxmisc/flowless/VirtualFlowContent.java]: And likewise this one as

    orientation.verticalPositionProperty(content)

File: src/main/java/org/fxmisc/flowless/VirtualFlowContent.java (hunk 1 above)
Comment (JordanMartinez) [inline src/main/java/org/fxmisc/flowless/VirtualFlowContent.java]: Mm... keeping the naming consistent.

Comment (JordanMartinez) [inline src/main/java/org/fxmisc/flowless/VirtualFlowContent.java]: Oh, I misread what you meant. Smh...

**Comment by TomasMikula** [comment]
Looks good to me now. Do you want to propose some method name changes?

**Comment by JordanMartinez** [comment]
Cool! Thank you for guiding me through all this.
The only proposition I have is changing the scrolling method names to be clearer.
There are the two methods now `scrollX(deltaX)` and `scrollXToPixel(pixel)` that can be misunderstood by a person who just starts using the library. If the first method's name was changed to `scrollXBy(deltaX)`, I think that would be clearer overall. The 2nd method could be renamed to `scrollXTo(pixel)`, but I don't like that for some reason. I'm not sure why.

**Comment by TomasMikula** [comment]
Makes sense. Though I would keep and deprecate the old `scrollX` and `scrollY` methods to give users some transition period. I cannot currently think of better names.

**Comment by JordanMartinez** [comment]
Ok. I'll also update OrientationHelper's method to correspond to the method naming.
scrollHorizontally & scrollVertically will become scrollHorizontallyBy & scrollVerticallyBy, respectively.

**Comment by TomasMikula** [comment]
Sounds good.

**Comment by JordanMartinez** [comment]
scrollX & scrollY are used in VirtualFlow's ScrollEventHandler, so I'll be updating those methods as well.

**Comment by TomasMikula** [comment]
Good.

In OrientationHelper, you don't need to keep any deprecated methods, because OrientationHelper is not part of public API.

Only keep and deprecate the old methods on VirtualFlow. Also, please use the [`@deprecated` Javadoc tag](http://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@deprecated) to explain the deprecation:

```java
/**
 * ...
 * @deprecated use {@link #scrollXBy(double)} instead.
 */
```

**Comment by JordanMartinez** [comment]
Still learning how to use those tags properly. I guess that's the next thing I should study.
I've updated it.

**Comment by TomasMikula** [comment]
Thanks, looks good. Merging.

**Comment by JordanMartinez** [comment]
Sweet! Thanks for guiding me through that.

To release this updated version, what else needs to be done? Now that the scrolling API is exposed, I can't expose it in RichTextFX until the new release, right?

**Comment by TomasMikula** [comment]
That is basically correct, although in your IDE you can set your local Flowless project as a dependency to your local RichTextFX project if you want to start playing with it.

I will release a new minor version soon.

**Comment by JordanMartinez** [comment]
Ok, cool.
True. I get that I need to change gradle's dependency management to use the local directory, but I haven't yet researched that enough to know how to do that.
Even so, I've already written a few lines of code (commented until I can figure out the dependency thing).

**Comment by TomasMikula** [comment]
At least Eclipse allows me to add projects as dependencies to other projects. No need to update gradle build script. (Of course that will only have effect in Eclipse, not on gradle builds.)

**Comment by TomasMikula** [comment]
Version 0.4.6 published.

**Comment by JordanMartinez** [comment]
Now that I think about it, I've already done something like that in my IDE in the past. Hmm... not sure why that didn't come to mind...

Sweet!

**Pull request merged**